Source/DafnyCore/DafnyOptions.cs

```diff
@@ -15,28 +19,87 @@
 using Bpl = Microsoft.Boogie;
 
 namespace Microsoft.Dafny {
+  public enum FunctionSyntaxOptions {
+    Version3,
+    Migration3To4,
+    ExperimentalTreatUnspecifiedAsGhost,
+    ExperimentalTreatUnspecifiedAsCompiled,
+    ExperimentalPredicateAlwaysGhost,
+    Version4,
+  }
+
+  public enum QuantifierSyntaxOptions {
+    Version3,
+    Version4,
+  }
 
-  public record Options(IDictionary<IOptionSpec, object> OptionArguments);
+  public record Options(IDictionary<Option, object> OptionArguments);
 
   public class DafnyOptions : Bpl.CommandLineOptions {
+
+    public void ApplyBinding(Option option) {
+      if (bindings.ContainsKey(option)) {
+        bindings[option](this, Get(option));
+      }
+    }
+
+    public T Get<T>(Option<T> option) {
+      return (T)Options.OptionArguments.GetOrCreate(option, () => default(T));
```

Comment:
Wouldn't it be better to use `() => option.GetDefaultValue()` instead?

Reply:
Yes but that doesn't exist. You can only get the default value by doing an actual parse of command line options. Maybe I can remove the default case here though.

Reply:
Ah it's quite possible I misread the `System.CommandLine` API.